api/src/services/user-db.js

```diff
@@ -26,24 +28,31 @@ const ddoc = {
   }
 };
 
-const createDb = (dbName, callback) => {
-  db.db.create(dbName, callback);
-};
-
-const setSecurity = (dbName, username, callback) => {
-  db.request({
-    db: dbName,
-    path: '/_security',
-    method: 'PUT',
-    body: {
-      admins: { names: [ username ], roles: [] },
-      members: { names: [ username ], roles:[] }
-    }
-  }, callback);
-};
-
-const putDdoc = (dbName, callback) => {
-  db.use(dbName).insert(ddoc, callback);
+const setSecurity = (dbName, username) => {
+  return new Promise((resolve, reject) => {
+    request.put({
```

Comment:
If you really want to swing for the fences you could replace `request` with `request-promises-native`

Reply:
I've been avoid adding more dependencies. I tried using util.promisify but it always stuffs up the mocking so I backed out.

I think you're right - we use request in enough places that it's worth the additional dependency. I'll do that in a different commit...